### DISCUSSION TITLE
Android: Use same formatting setup for Android Studio formatting and Detekt

### DISCUSSION THREAD

**Discussion by MarcLFJ** [detekt/detekt#4663] [Q&A] [answered]
I have an Android project that i've set up with Detekt.
I'm currently using:
`io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.19.0` (Root build.gradle)
`io.gitlab.arturbosch.detekt:detekt-formatting:1.19.0` (in my modules).
Running `gradle detekt` works fine and it's correctly formats code in regards to ktlint rules and I've added logic to run detekt on compile successfully, with the built in ktlint rules in detekt-formatting. :)
My question is: I can't seem to find anything about how to align the auto formatting in Android Studio (Code > Reformat file), which, in my case, is set to be a specific one for my project `({root}/.idea/codeStyles/...)`, with the formatting used by Detekt.
Reason for wanting this, is that i now have to maintain two formatting style files and keep them aligned, instead of being able to make both formatters point to the same config file. 

Please let me know if this is the wrong place for this or i need to supply more info!

**Comment by schalkms** [comment, marked as answer]
There exists no guide on how to align `detekt-formatting` with IDEA formatting, to my knowledge. 
There are two solutions that I can think of.

1. You decide for one or the other formatting tool.
Arguably, those formatting tools differ quite a bit starting with the configuration file. Furthermore, I think it's rather a seldom use case to have two formatting tools in the same project.

2. You write a [custom 3rd party detekt ruleset](https://detekt.dev/extensions.html#custom-rulesets) that reads the IDEA config file and configures the formatting rules of detekt according to the xml attributes you need.
`detekt-formatting` wraps ktlint rules, which can be configured in detekt but not in *plain* ktlint. You could take a look at the implementation in the main repo. This [package](https://github.com/detekt/detekt/tree/main/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting) would be a good starting point.
